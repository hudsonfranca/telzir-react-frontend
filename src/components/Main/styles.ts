import styled from 'styled-components';

export const Container = styled.main`
  grid-area: main;
  position: relative;
`;

export default Container;
