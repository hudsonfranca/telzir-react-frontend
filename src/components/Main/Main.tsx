import React from 'react';

import { Container } from './styles';

const Main: React.FC = ({ children }) => {
  return <Container data-test="mainContainer">{children}</Container>;
};

export default Main;
