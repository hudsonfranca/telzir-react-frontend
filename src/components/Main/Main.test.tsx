import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import Main from './Main';
import 'jest-styled-components';

let wrapper: ShallowWrapper;

beforeEach(() => {
  wrapper = shallow(<Main />);
});

describe('<Layout /> redering', () => {
  it('Should render without errors', () => {
    const main = renderer.create(<Main />).toJSON();

    expect(main).toMatchSnapshot();
  });

  it('Should render 1 mainContainer', () => {
    expect(wrapper.find(`[data-test='mainContainer']`).length).toEqual(1);
  });
});
