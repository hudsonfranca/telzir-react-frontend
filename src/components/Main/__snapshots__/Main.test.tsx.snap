// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> redering Should render without errors 1`] = `
.c0 {
  grid-area: main;
  position: relative;
}

<main
  className="c0"
  data-test="mainContainer"
/>
`;
